ydb/core/http_proxy/json_proto_conversion.h

```diff
@@ -147,59 +147,78 @@ inline void AddJsonObjectToProtoAsMap(
         const google::protobuf::Reflection* reflection,
         grpc::protobuf::Message* message,
         const JSON& jsonObject,
+        ui32 depth,
         std::function<const MAP(const JSON&)> extractMap,
-        std::function<const TString(const JSON&)> valueToString
+        std::function<const TString(const JSON&)> valueToString,
+        std::function<void(const JSON&, grpc::protobuf::Message*, ui32)> jsonObjectToMessage
 ) {
     const auto& protoMap = reflection->GetMutableRepeatedFieldRef<google::protobuf::Message>(message, fieldDescriptor);
     for (const auto& [key, value] : extractMap(jsonObject)) {
-        std::unique_ptr<google::protobuf::Message> stringStringEntry(
+        std::unique_ptr<google::protobuf::Message> mapEntry(
             google::protobuf::MessageFactory::generated_factory()
                 ->GetPrototype(fieldDescriptor->message_type())
                 ->New(message->GetArena())
         );
-        stringStringEntry
+        mapEntry
             ->GetReflection()
-            ->SetString(stringStringEntry.get(), fieldDescriptor->message_type()->field(0), key);
-        stringStringEntry
-            ->GetReflection()
-            ->SetString(stringStringEntry.get(), fieldDescriptor->message_type()->field(1), valueToString(value));
-        protoMap.Add(*stringStringEntry);
+            ->SetString(mapEntry.get(), fieldDescriptor->message_type()->field(0), key);
+
+        auto valueField = fieldDescriptor->message_type()->field(1);
+        if (valueField->cpp_type() == google::protobuf::FieldDescriptor::CPPTYPE_MESSAGE) {
+            auto *msg = mapEntry->GetReflection()->MutableMessage(mapEntry.get(), valueField);
+            jsonObjectToMessage(value, msg, depth);
+        } else if (valueField->cpp_type() == google::protobuf::FieldDescriptor::CPPTYPE_STRING) {
+            mapEntry->GetReflection()->SetString(mapEntry.get(), valueField, valueToString(value));
+        } else {
```

Comment:
Ideally, we should support all types of map keys and reuse existing code for it, but for now we only use String and Message and I think it's ok to leave it as is temporarily.